eras/conway/impl/src/Cardano/Ledger/Conway/Genesis.hs

```diff
@@ -12,31 +13,70 @@ where
 import Cardano.Ledger.Binary (
   DecCBOR (..),
   EncCBOR (..),
+  decodeNullStrictMaybe,
+  encodeNullStrictMaybe,
  )
+import Cardano.Ledger.Conway.Governance
 import Cardano.Ledger.Conway.PParams (UpgradeConwayPParams)
-import Cardano.Ledger.Crypto (Crypto)
-import Data.Aeson (FromJSON (..), ToJSON)
-import Data.Aeson.Types (ToJSON (..))
+import Cardano.Ledger.Core
+import Data.Aeson (
+  FromJSON (..),
+  KeyValue (..),
+  ToJSON (..),
+  object,
+  pairs,
+  withObject,
+  (.:),
+  (.:?),
+ )
+import Data.Default.Class (Default (def))
 import Data.Functor.Identity (Identity)
+import Data.Maybe.Strict (StrictMaybe, maybeToStrictMaybe)
 import GHC.Generics (Generic)
 import NoThunks.Class (NoThunks)
 
-newtype ConwayGenesis c = ConwayGenesis
-  { cgUpgradePParams :: UpgradeConwayPParams Identity
+data ConwayGenesis c = ConwayGenesis
+  { cgUpgradePParams :: !(UpgradeConwayPParams Identity)
+  , cgConstitution :: !(Constitution c)
+  , cgCommittee :: !(StrictMaybe (Committee c))
```

Comment:
Genesis type only makes sense for the era it is defined for. Also `c` is not `era` :wink::

Also, we will have to provide the initial committe, so it should not be optional
```suggestion
data ConwayGenesis c = ConwayGenesis
  { cgUpgradePParams :: !(UpgradeConwayPParams Identity)
  , cgConstitution :: !(Constitution (ConwayEra c))
  , cgCommittee :: !(Committee (ConwayEra c))
```